sql/core/src/test/resources/sql-tests/results/udf/udf-window.sql.out

```diff
@@ -154,17 +154,17 @@ SELECT val_timestamp, udf(cate), avg(val_timestamp) OVER(PARTITION BY udf(cate)
 RANGE BETWEEN CURRENT ROW AND interval 23 days 4 hours FOLLOWING) FROM testData
 ORDER BY udf(cate), val_timestamp
 -- !query schema
-struct<val_timestamp:timestamp,CAST(udf(cast(cate as string)) AS STRING):string,avg(CAST(val_timestamp AS DOUBLE)) OVER (PARTITION BY CAST(udf(cast(cate as string)) AS STRING) ORDER BY val_timestamp ASC NULLS FIRST RANGE BETWEEN CURRENT ROW AND INTERVAL '23 days 4 hours' FOLLOWING):double>
+struct<val_timestamp:timestamp,CAST(udf(cast(cate as string)) AS STRING):string,avg(val_timestamp) OVER (PARTITION BY CAST(udf(cast(cate as string)) AS STRING) ORDER BY val_timestamp ASC NULLS FIRST RANGE BETWEEN CURRENT ROW AND INTERVAL '23 days 4 hours' FOLLOWING):timestamp>
 -- !query output
 NULL	NULL	NULL
-2017-07-31 17:00:00	NULL	1.5015456E9
-2017-07-31 17:00:00	a	1.5016970666666667E9
-2017-07-31 17:00:00	a	1.5016970666666667E9
-2017-08-05 23:13:20	a	1.502E9
-2020-12-30 16:00:00	a	1.6093728E9
-2017-07-31 17:00:00	b	1.5022728E9
-2017-08-17 13:00:00	b	1.503E9
-2020-12-30 16:00:00	b	1.6093728E9
+2017-07-31 17:00:00	NULL	2017-07-31 17:00:00
+2017-07-31 17:00:00	a	2017-08-02 11:04:26.666666
+2017-07-31 17:00:00	a	2017-08-02 11:04:26.666666
+2017-08-05 23:13:20	a	2017-08-05 23:13:20
+2020-12-30 16:00:00	a	2020-12-30 16:00:00
+2017-07-31 17:00:00	b	2017-08-09 03:00:00
+2017-08-17 13:00:00	b	2017-08-17 13:00:00
+2020-12-30 16:00:00	b	2020-12-30 16:00:00
```

Comment:
Hm .. so we allow timestamp types whereas other DBMSes disallow?

Reply:
Well, there isn't a real consensus around other DBMS'es. Keeping it a Timestamp seems like something that you would expect. MySQL's behavior is much more awkward in my opinion. Spark needs to pave the path on this one :)

Reply:
It seems Oracle cannot accept timestamp for average, too. Any other system supporting this behaivour?